core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -610,6 +611,13 @@ public boolean isReady() {
       }
       return realStream.isReady();
     }
+
+    @Override
+    public void setMessageCompression(boolean enable) {
+      if (realStream != null) {
+        realStream.setMessageCompression(enable);
```

Comment:
You are right.  I have updated this to be properly delayed.  

Note: there is a fast path available with the volatile realStream, but i figured it wasn't worth the special casing logic.  Nothing is going out over the network, so the latency is already going to be high.

@ejona86 PTAL